src/name_utils.rs

```diff
@@ -0,0 +1,399 @@
+//! The "Stackable" way to implement our Kubernetes resource names.
+//!
+//! We follow the specification for RFC 1035 names (https://tools.ietf.org/html/rfc1035):
+//! This means the name must:
+//! * contain at most 63 characters
+//! * contain only lowercase alphanumeric characters or '-'
+//! * start with an alphabetic character
+//! * end with an alphanumeric character
+//!
+//! The Stackable name is structured in the following pattern ("(<...>)" means optional):
+//! <short-name>-<cluster-name>-<role_name>-(<group_name>)-(<node_name>)-(<misc>)-
+//!
+//! Additionally all non alphanumeric characters are removed.
+//!
+//! This generated name is supposed to work with the generatedName (to add a unique hash) from Kubernetes.
+//!
+use crate::error::OperatorResult;
+
+/// Adjustable number of characters reserved for a generated Kubernetes hash.
+const KUBERNETES_HASH_MIN_LENGTH: usize = 8;
+/// This is the overall fixed length for resource names.
+/// WARNING: Do not change this number unless any specifications change.
+const RESOURCE_NAME_MAX_LEN: usize = 63;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
Just wondering why these are not in error.rs.

Reply:
I like to have fewer operator errors and derive from smaller errors. Its just a preference, no real difference. Since its not an external library it would make sense in the main errors though.